src/gui_common/CustomDropDown.cs

```diff
@@ -0,0 +1,165 @@
+using System.Collections.Generic;
+using Godot;
+
+/// <summary>
+///   Basically a class that inherits MenuButton, but with extra popup menu functionality
+///   such as adjusted custom icon size with tweakable color and some slide down animation.
+///   (Might need to expand this later)
+/// </summary>
+public class CustomDropDown : MenuButton
+{
+    /// <summary>
+    ///   The MenuButton's popup menu
+    /// </summary>
+    public PopupMenu Popup;
+
+    private Tween tween;
+
+    /// <summary>
+    ///   All item icon sizes will be adjusted according to this. Currently it's automatically
+    ///   set according to the PopupMenu's check icon size (with a bit smaller result)
+    /// </summary>
+    private Vector2 iconSize;
+
+    private float cachedPopupHSeparation;
+    private float cachedPopupVSeparation;
+
+    private List<Item> items;
+
+    public CustomDropDown()
+    {
+        Popup = GetPopup();
+        items = new List<Item>();
+        tween = new Tween();
+
+        AddChild(tween);
+
+        cachedPopupHSeparation = Popup.GetConstant("hseparation");
+        cachedPopupVSeparation = Popup.GetConstant("vseparation");
```

Comment:
Are these not available through normal properties?

Reply:
They could be made into properties

Reply:
I don't see a huge need for that, I meant as in, does Godot not give this information through properties? I'm pretty sure margins etc. can be fetched through properties. But not the hseparation?

Reply:
No, I think they have to be fetched that way because custom constants varies depending on the control derived types, so like the MarginContainer has margin constants and VBoxContainer has separation constant. Wouldn't make sense for the Control base class has them as properties.

Reply:
But you are using a PopupMenu class here? Though, it looks like that is indeed missing access to the hseparation.

Reply:
Yeah, `GetConstant()` have to be used to retrieve Theme properties afaik